lib/Runtime/Library/JsBuiltInEngineInterfaceExtensionObject.cpp

```diff
@@ -175,12 +177,46 @@ namespace Js
         case PropertyIds::String:
             return library->stringPrototype;
 
+        case PropertyIds::__chakraLibrary:
+            return library->GetChakraLib();
+
         default:
             AssertMsg(false, "Unable to find a prototype that match with this className.");
             return nullptr;
         }
     }
 
+    Var JsBuiltInEngineInterfaceExtensionObject::EntryJsBuiltIn_RegisterChakraLibraryFunction(RecyclableObject* function, CallInfo callInfo, ...)
+    {
+        EngineInterfaceObject_CommonFunctionProlog(function, callInfo);
+
+        Assert(args.Info.Count >= 3 && JavascriptString::Is(args.Values[1]) && JavascriptFunction::Is(args.Values[2]));
```

Comment:
I know this is called only internally, but we probably should do runtime checks of the number of arguments and their types